### PR TITLE
Fix item upgrade modifier persist on death

### PR DESCRIPTION
**Pull request by carlosrpg**

### CODE REVIEW

File: game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua [hunk 1]
```diff
@@ -8,8 +8,12 @@ function modifier_aura_item_upgrade:IsPurgable()
   return false
 end
 
-function modifier_aura_item_upgrade:GetAttributes()
-  return MODIFIER_ATTRIBUTE_MULTIPLE
```
Comment (Trildar) [inline game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua]: Why was this removed? I'm pretty sure it is 100% necessary. Without it, it won't allow multiple instances of the modifier on the hero and it will probably not work properly when items are combined in quick succession.

File: game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua [hunk 2]
```diff
@@ -50,6 +54,9 @@ if IsServer() then
         end
       end
     end
+
+    self:Destroy()
```
Comment (Trildar) [inline game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua]: Was this tested? I would think it would cause the modifier to destroy itself before it has a chance to run its think function.

Comment (Trildar) [reply game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua]: Oh. Duh. It returns early when it finds an item.

File: game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua (hunk 1 above)
Comment (carlosrpg) [reply game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua]: You are right

File: game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua (hunk 2 above)
Comment (carlosrpg) [reply game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua]: Without the DestroyOnExpire it was getting destroyed after the first think

File: game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua (hunk 1 above)
Comment (chrisinajar) [reply game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua]: Truth, I only tested with 1 item at a time.. Easiest way to test is get everything ready for combining in stash then walk into the other well. Items will instant combine and you die a quarter second later.
